core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -514,6 +516,14 @@ private void validate(CreateIndexClusterStateUpdateRequest request, ClusterState
     }
 
     public void validateIndexSettings(String indexName, Settings settings) throws IndexCreationException {
+        List<String> validationErrors = indexSettingsValidation(settings);
+        if (validationErrors.isEmpty() == false) {
+            throw new IndexCreationException(new Index(indexName),
+                    new IllegalArgumentException(getMessage(validationErrors)));
+        }
+    }
+
+    public List<String> indexSettingsValidation(Settings settings) {
```

Comment:
maybe this method could be package private? Also, I would prefer a different name, like `getIndexSettingsValidationErrors()` for instance.